pybop/costs/base_cost.py

```diff
@@ -96,6 +104,9 @@ def __call__(
         else:
             model_inputs = inputs
 
+        # Check whether we are maximising or minimising
```

Comment:
This took me a second on first pass, to save headache in the future adding this table would be helpful:
```suggestion
        # Check whether we are maximising or minimising via:
        #| `minimising` | `self.minimising` | `for_optimiser` |
        #|--------------|-------------------|-----------------|
        #| `True`       | `True`            | `True`          |
        #| `True`       | `True`            | `False`         |
        #| `False`      | `False`           | `True`          |
        #| `True`       | `False`           | `False`         |
```